Delcam.ProductInterface.PowerMILL/Classes/PMProject.cs

```diff
@@ -430,7 +430,10 @@ public void CreateBlock(string modelName, double blockExpansion)
         /// <param name="boundary">The boundary from which to create the block.</param>        
         public BoundingBox CreateBlockFromBoundary(PMBoundary boundary)
         {
-            if (boundary != null)
+            if (boundary == null)
+                throw new ArgumentNullException("boundary", "Boundary not found");
+
+            if (boundary.Exists)
```

Comment:
At this point, I think you should combine the two.  Have:
`if (boundary == null || !boundary.Exists)`